src/components/object-info/ObjectInfo.js

```diff
@@ -74,9 +74,16 @@ const ObjectInfo = ({ t, tReady, className, type, cid, localPath, size, data, li
           <a className='dn di-ns link charcoal ml2' href='https://docs.ipfs.io/concepts/file-systems/#unix-file-system-unixfs' target='_external'>UnixFS</a>
         ) : null}
         {format === 'unixfs' && data.type && ['directory', 'file'].some(x => x === data.type) ? (
-          <a className='link avenir ml2 pa2 fw5 f6 blue' href={`${gatewayUrl}/ipfs/${cid}`} target='_external'>
-            {t('ObjectInfo.gatewayLink')}
-          </a>
+          <>
+            {gatewayUrl && gatewayUrl !== publicGatewayUrl && (
+              <a className='link avenir ml2 pa2 fw5 f6 blue' href={`${gatewayUrl}/ipfs/${cid}`} target='_external'>
+                {t('ObjectInfo.privateGateway')}
+              </a>)}
+            {publicGatewayUrl && (
+              <a className='link avenir ml2 pa2 fw5 f6 blue' href={`${publicGatewayUrl}/ipfs/${cid}`} target='_external'>
```

Comment:
We may want to add `#x-ipfs-companion-no-redirect` to the end of the public gateway link in case when a local one is present.
Rationale: this usually means user has ipfs-companion and public gateway links get redirected to local one anyway, so we need to explicitly disable that redirect.